File: apps/mybookkeeper/backend/app/schemas/leases/signed_lease_import_request.py
"""Schema for POST /signed-leases/import — upload externally-signed PDFs."""
from __future__ import annotations

import datetime as _dt
import uuid

from pydantic import BaseModel, ConfigDict, Field


class SignedLeaseImportRequest(BaseModel):
    applicant_id: uuid.UUID
    listing_id: uuid.UUID | None = None
    starts_on: _dt.date | None = None
    ends_on: _dt.date | None = None
    notes: str | None = Field(default=None, max_length=2000)
    # Default to 'signed' — by definition, imported leases are already signed.
    status: str = "signed"

    model_config = ConfigDict(extra="forbid")

File: apps/mybookkeeper/frontend/src/shared/types/lease/signed-lease-import-request.ts
export interface SignedLeaseImportRequest {
  applicant_id: string;
  listing_id?: string;
  starts_on?: string;
  ends_on?: string;
  notes?: string;
  status?: string;
  files: File[];
  /** Optional successor pointer (same semantics as the create flow). */
  parent_lease_id?: string;
}
